src/main/java/com/sixt/service/framework/rpc/HttpClientWrapper.java

```diff
@@ -163,6 +163,9 @@ public ContentResponse execute(HttpRequestWrapper request, RpcCallExceptionDecod
                     }
                 }
                 if (tryCount < client.getRetries()) {
+                    if (client.hasRetryBackOffFunction()) {
+                        client.getRetryBackOffFunction().execute(tryCount);
```

Comment:
Hi @boyand and @bkrahmer , by chance I came across this PR and had a look at the code.

First I am not a Java expert but as I understand the code I think that there might be some errors in the behaviour of the RetryBackOff implementation. It introduces a concurrent dependency [here](https://github.com/Sixt/ja-micro/pull/51/files#diff-b0edadd4361c5c71c87e93e089e75c7dR28) when it is invoked by different threads, that I think might stack the wait time. I think that this concurrent dependency is actually not required. 

Couldn't the whole implementation be done by using something like: 
```java
Thread.sleep(client.getRetryBackOffDuration().duration(tryCount))
```
instead of having the largish `execute` function in the interface? So the `RetryBackOffFunction` interface could be simplified to just keep the `timeout` definition. 

The `ExponentialRetryBackOff` is just one implementation of the `getRetryBackOffFunction` and could be initialised somewhere as default value in the class, if this is desired.

WDYT about this?